src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -85,17 +84,24 @@ func (cs *ChallengeStatus_I) IsChallenged() bool {
 	return cs.LastChallengeEpoch() > cs.LastChallengeEndEpoch()
 }
 
-func (st *StorageMinerActorState_I) _isChallenged(rt Runtime) bool {
-	return st.ChallengeStatus().IsChallenged()
-}
-
 func (a *StorageMinerActorCode_I) _isChallenged(rt Runtime) bool {
 	h, st := a.State(rt)
 	ret := st._isChallenged(rt)
 	Release(rt, h, st)
 	return ret
 }
 
+func (cs *ChallengeStatus_I) ShouldChallenge(currEpoch block.ChainEpoch, minChallengePeriod block.ChainEpoch) bool {
+	// true when currEpoch > LastChallengeEpoch + minChallengePeriod
+	return currEpoch > (cs.LastChallengeEpoch() + minChallengePeriod)
+}
+
+func (st *StorageMinerActorState_I) _shouldChallenge(rt Runtime, networkPower block.StoragePower) bool {
+	// pulled in from constants
```

Comment:
Let's define this in constant then and we can define this in ChallengeStatus.ShouldChallenge and this method will just be a pass-through.